quinn-proto/src/transport_parameters.rs

```diff
@@ -467,6 +475,81 @@ impl TransportParameters {
     }
 }
 
+/// A reserved transport parameter.
+///
+/// It has identifier of the form 31 * N + 27 for integer value of N.
+/// Such identifiers reserved to exercise the requirement that unknown transport parameters be ignored.
+/// The reserved transport parameter have no semantics and can carry arbitrary values.
+/// It may be included into quic_transport_parameters TLS extensions when sent and should be ignored when received.
+///
+/// See spec: <https://www.rfc-editor.org/rfc/rfc9000.html#section-18.1>
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub(crate) struct ReservedTransportParameter {
+    /// The reserved identifier of transport parameter
+    id: VarInt,
+
+    /// Buffer to store parameter payload
+    payload: [u8; Self::MAX_PAYLOAD_LEN],
+
+    /// The number of bytes to include into wire format from the `payload` buffer
+    payload_len: usize,
+}
+
+impl ReservedTransportParameter {
+    /// The maximum length of payload to include as parameter payload.
+    /// This value is not a specification-imposed limit and picked such that
+    /// it matches the limit by other implementations of quic, e.g. quic-go & quiche.
+    const MAX_PAYLOAD_LEN: usize = 16;
+
+    // Generate a random reserved identifier of the form `31 * N + 27` as required by RFC 9000.
+    // Reserved transport parameter identifiers are used to test compliance with the requirement
+    // that unknown transport parameters be ignored by peers.
+    // See: <https://www.rfc-editor.org/rfc/rfc9000.html#section-18.1> and <https://www.rfc-editor.org/rfc/rfc9000.html#section-22.3>
+    fn generate_reserved_id<R: rand::RngCore>(rng: &mut R) -> VarInt {
```

Comment:
Nit: move `generate_reserved_id()` below `random()` (since the latter depends on the former, and same point about taking the generic parameter for all three of the associated functions here.